Login view and other small changes

BudsieBuds opened the pull request:
Hi everyone,

<details>
<summary>Read this for some context</summary>
Thought I'd provide a little context to what, how and why for who wants to know. So I bumped into this repository while looking for a POS system for a friend's store. That's when I decided to dive into the code, mainly for coding practice. Full disclosure: I'm not a developer, just trying to learn stuff. I've put a lot of time in this pull request and I learned a lot while coding it. It's the first time I create a pull request of this scale, so please be mild. 😉 Probably better to create a pull request per change next time (?). This seems a bit messy, sorry for that. I'll try my best to explain everything below. Also, English is not my native language. Therefore, if anything is unclear, please give me a chance to provide additional info. Thanks!
</details>

## :tv: Login view
* Updated to more modern style and coding standards.
* Login screen fully respects theme colors.
* Fully responsive. Looks good on all displays I've tested. From small screens like iPhone SE to big 4K monitors.
* Uses Bootstrap 5. Should be included in addition to Bootstrap 3.
* Uses Bootswatch 5. Should be included in addition to Bootswatch 3.
  * Fallback for unsupported BS3 themes to Flatly included.
* Shows company logo when available.
  * Fallback to OSPOS logo.
* Added option to `config/general` to change login form style. Two options: floating labels or input groups.
* Added default OSPOS color as `theme-color` for Chrome Android.

## :mouse: Smaller changes
* Capitalize theme names in theme selector.
* Added theme preview link to `config/general` screen.
* Shifted around some translations for easier and more consistent translating.

## :mailbox_with_mail: Suggestions/Questions/Discussion
* Should the version number be removed from the login title and footer for security reasons?
* Is there an SVG version of the OSPOS logo available? I've auto converted it for testing purposes.
* Made the OSPOS logo coloring and rounding to match the selected theme. Don't know if this is something you're ok with? I think it makes the logo look fitting, and good visibility is guaranteed on all themes.
* Can SVG be enabled for company logo upload or will this give errors with logo use in other parts of the POS?
* I'd like the default Bootstrap theming to be added in addition to the Bootswatch themes.
* Should there be an option to add an Apple touch icon? Maybe in an advanced config tab. Could also be moved to a separate `config/design` together with the theme selector and the login form style option.
* Added `noindex, nofollow` to the login view. Is there an argument not too?
* Should there be an additional branch for more modern browsers or rather a legacy branch for older browsers? Don't know if this is needed, IE11 goes EOL on 15th of June 2022 and support from frameworks for older browser versions decreases. For me personally, there isn't a use case where modern browsers can't be used. Are there any or have you decided this, maybe based on browser usage stats?

## :page_facing_up: Notes
* Tested everything on a local installation and applied the following changes there:
  * Renamed `public/dist/bootswatch` folder to `public/dist/bootswatch-3`.
  * Added `public/dist/bootswatch-5` folder with the latest Bootswatch 5 themes.
  * Added `public/dist/bootswatch-3/bootstrap` and `public/dist/bootswatch-5/bootstrap` for default Bootstrap theming.
* The `bower.json` and `Gruntfile.js` are examples! Don't think they will work as provided.
* Addition of `OSPOS` and `Open Source Point of Sale` to `common_lang.php` should probably be set to read-only.

Thanks for reading and I hope I made something that can be implemented. In the meanwhile I'll be trying to convert the rest of the interface to more modern standards.

Greetings,
Bas ~ BudsieBuds

_**edit:** added some screenshots_
| Screenshot 1 | Screenshot 2 | Screenshot 3 |
| - | - | - |
| <img src="https://user-images.githubusercontent.com/12870258/119424848-8b552a80-bd06-11eb-92d5-9381161807c5.png" data-canonical-src="https://user-images.githubusercontent.com/12870258/119424848-8b552a80-bd06-11eb-92d5-9381161807c5.png" height="250px" /> | <img src="https://user-images.githubusercontent.com/12870258/119424850-8bedc100-bd06-11eb-9a01-1617ab2328dc.png" data-canonical-src="https://user-images.githubusercontent.com/12870258/119424850-8bedc100-bd06-11eb-9a01-1617ab2328dc.png" height="250px" /> | <img src="https://user-images.githubusercontent.com/12870258/119424851-8c865780-bd06-11eb-8068-e0fceb0b7b36.png" data-canonical-src="https://user-images.githubusercontent.com/12870258/119424851-8c865780-bd06-11eb-8068-e0fceb0b7b36.png" height="250px" /> |
| Screen size: iPad | Screen size: iPhone 8 Plus | Screen size: iPhone SE |
| Theme: Flatly | Theme: Cosmo | Theme: Darkly |
| Login Form: Floating | Login Form: Floating | Login Form: Input Groups |

jekkos commented:
This looks really great, @BudsieBuds . We have a ticket to migrate to a newer bootstrap #841 . Currently we don't have any good frontend devs contributing so this one is really welcomed. I'll review a bit more in depth later this week, but this certainly looks promising.

BudsieBuds commented:
Thanks @jekkos! Let me know if anything is unclear.

daN4cat commented:
Removing the version from the login page would be a good idea indeed from a security perspective l.
Thanks for the contribution and good work.

jekkos commented on Gruntfile.js:
```diff
@@ -40,13 +40,22 @@ module.exports = function(grunt) {
+					destPrefix: 'public/dist'
+				},
+				files: {
+					bootswatch-5: '*/'
```
as per your comments, these are examples so these files are probably not minified by the build?

BudsieBuds replied:
I just can't get get stuff like docker, bower, composer and grunt to work on my Windows desktop. Also haven't got any experience with these tools, so that's probably part of the problem. My knowledge and Google's couldn't help me out here. Spent a few evenings trying to get it to work for a better dev experience, but to no avail. I've been developing via a local XAMPP installation, this makes the coding pretty tedious. Honestly there's a lot I don't understand how and why it happens and have to figure stuff out as I go. That's why I edited this and the bower file by hand as an example.

jekkos replied:
Ok it might be a pain on windows so maybe its better to run everything natively

BudsieBuds replied:
Any tips on how to set this up?

jekkos replied:
I don't really have a lot of experience running docker on windows.. But once you have docker compose running then you should be able to run the dev container just fine?

There should be some tutorials on this somewhere.

jekkos replied:
Currently the bower does seem to crash on the bootswatch-5 download.

jekkos replied:
To install the two versions simulatenously, you'll need to do something `[like this](https://stackoverflow.com/questions/16442012/bower-install-2-versions-of-jquery)`

jekkos replied:
Ok I think what you did is correct according to those docs, I explained the issue a bit further below.

jekkos commented on application/views/login.php:
```diff
@@ -1,54 +1,101 @@
 	<!-- start css template tags -->
-	<link rel="stylesheet" type="text/css" href="css/login.css"/>
+	<link rel="stylesheet" type="text/css" href="css/login.min.css">
+  <link rel="stylesheet" type="text/css" href="css/rounding.min.css">
```
I think you can upgrade the login page to the newer bootstrap only for now. It does not use the shared header with all the imports that the main application uses, so from that point of view, it should be fine to import bootstrap5 here only for now.

BudsieBuds replied:
The current login view also uses the Bootswatch files though.

jekkos replied:
Correct, I might have overlooked that. We need to keep them separated then for the time being.
I do think that the login page is not using Grunt directly so maybe you can just refer to the theme once it is downloaded by bower (so no need to adapt the GruntFile or minification build)?

jekkos commented on application/views/login.php:
```diff
@@ -1,54 +1,101 @@
+        <svg class="logo text-primary rounded--<?php echo $this->config->item('theme'); ?>" xmlns="http://www.w3.org/2000/svg" role="img" version="1.0" viewBox="0 0 90 90">
+          <title><?php echo $this->lang->line('common_logo') . '&nbsp;' . $this->lang->line('common_software_short'); ?></title>
+          <rect fill="currentColor" width="100%" height="100%"/>
+          <path fill="white" d="M58.4 17c-3.8.1-8.3.7-10.1 1.5-1.7.7-4.1 2.3-5.3 3.5l-2.1 2.1-2.6-2.1c-6.4-5-15.3-4.6-21.8 1.1-8.8 7.7-6.3 23.1 4.4 27.6A20.1 20.1 0 0038 49c3.8-2.7 6.9-9 7-14.2a12 12 0 017.8-10.3l2.4-.7-.4 12c-.3 11-.5 12.3-2.6 14.5a11.9 11.9 0 01-4.8 3.1c-2 .5-2.4 1.3-2.4 4.1 0 1.9.3 3.5.8 3.5 3.5 0 8.6-2.7 11.6-6 2.1-2.4 4.5-4 5.8-4 3.9-.1 10.4-3.6 12.8-7 3.7-5.1 3.9-12.8.6-18. ...
```
Thanks for converting this to svg, i suppose this is how one should design the logo in the end.

jekkos commented on application/views/login.php:
```diff
@@ -1,54 +1,101 @@
+        </div>
+        <div class="input-group mb-3">
+          <span class="input-group-text" id="input-password">
+            <svg xmlns="http://www.w3.org/2000/svg" height="1em" fill="currentColor" class="bi bi-key-fill" viewBox="0 0 16 16">
```
I guess it's fine to keep this in here but once we want to reuse it, then it might be better to move it to a separate view file (that can be included from others)

jekkos commented on application/views/login.php:
```diff
@@ -1,54 +1,101 @@
+      <span class="text-muted">
+        <svg xmlns="http://www.w3.org/2000/svg" version="1.0" height="1em" role="img" viewBox="0 0 67.772865 44">
+          <title><?php echo $this->lang->line('common_software_short'); ?></title>
+          <path fill="currentColor" d="M37.139292 1.5c-1.7.7-4.1 2.3-5.3 3.5l-2.1 2.1-2.6-2.1c-6.4-5-15.3-4.6-21.8 1.1-8.8 7.7-6.3 23.1 4.4 27.6 4.9 2.1 12.9 1.3 17.1-1.7 3.8-2.7 6.9-9 7-14.2 0-4.2 3.8-9.1 7.8-10.3l2.4-.7-.4 12c-.3 11-.5 12.3-2.6 14.5-1.3 1.4-3.5 2.8-4.8 3.1-2 .5-2.4 1.3-2.4 4.1 0 1.9.3 3.5.8 3.5 3.5 0 8.6-2.7 11.6-6 2.1-2.4 4.5-4 5.8-4 3.9-.1 10.4-3.6 12.8-7 3.7-5.1 3.9-12.8.6-1 ...
```
Is this like a parameterized version of the previous svg? Or isn't that really feasible to do?

BudsieBuds replied:
I can reuse the path from the previous SVG. I'll reevaluate this if I succeed in converting it to a better SVG.

jekkos replied:
Ok sounds great

jekkos commented:
> ##
> 
>     * Should the version number be removed from the login title and footer for security reasons?
> 
>     * Is there an SVG version of the OSPOS logo available? I've auto converted it for testing purposes.
Currently there was no svg version, so you're the first to add it.
> 
>     * Made the OSPOS logo coloring and rounding to match the selected theme. Don't know if this is something you're ok with? I think it makes the logo look fitting, and good visibility is guaranteed on all themes.
The screencaps you post here look good in fact.

> 
>     * Can SVG be enabled for company logo upload or will this give errors with logo use in other parts of the POS?

I don't think it's allowed for now. If we want to enable it then maybe let's do that in a separate PR
> 
>     * I'd like the default Bootstrap theming to be added in addition to the Bootswatch themes.
Sounds like a good idea. I must say that in this case it was easy to add bootswatch as it's downloaded by bower and then ready to use from the extracted directory. You can add a ticket specifically for this one so we can discuss the details there
> 
>     * Should there be an option to add an Apple touch icon? Maybe in an advanced config tab. Could also be moved to a separate `config/design` together with the theme selector and the login form style option.
I'm not really familiar with how apple touch works, but I guess iOs users might find it useful.

> 
>     * Added `noindex, nofollow` to the login view. Is there an argument not too?
Yes thanks for adding that. I suppose a web crawler won't get any further then the login page, but even then we might prefer not to have that one indexed
> 
>     * Should there be an additional branch for more modern browsers or rather a legacy branch for older browsers? Don't know if this is needed, IE11 goes EOL on 15th of June 2022 and support from frameworks for older browser versions decreases. For me personally, there isn't a use case where modern browsers can't be used. Are there any or have you decided this, maybe based on browser usage stats?

I don't think there is any specific demand for this currently. Do you know whether an upgrade to bootstrap 5 would break compatibility with some browsers?

jekkos commented:
> Removing the version from the login page would be a good idea indeed from a security perspective.
> Thanks for the contribution and good work.

But I would keep it in the footer once logged in

jekkos commented:
Final nitpick: regarding the git log, once done maybe we can try to summarize the changes in the issue in [the commit log](https://chris.beams.io/posts/git-commit/) as well?

BudsieBuds commented:
> I don't think it's allowed for now. If we want to enable it then maybe let's do that in a separate PR

Can confirm that it is not allowed. I'll put it on my list to look at later.

> Sounds like a good idea. I must say that in this case it was easy to add bootswatch as it's downloaded by bower and then ready to use from the extracted directory. You can add a ticket specifically for this one so we can discuss the details there

Probably better to point the theme selector to include the default location of the Bootstrap CSS. I don't think it's good practice to include it twice. Added it to my list for a separate PR.

> I'm not really familiar with how apple touch works, but I guess iOs users might find it useful.

I'll try and figure out what is the best practice for favicon, apple touch icon, etcetera and come up with an idea in a PR.

> I don't think there is any specific demand for this currently. Do you know whether an upgrade to bootstrap 5 would break compatibility with some browsers?

The main difference is that BS3 officialy supports IE8-11, BS4 supports IE10+ and BS5 drops support for IE completely.
[BS3](https://getbootstrap.com/docs/3.4/getting-started/#support) [BS4](https://getbootstrap.com/docs/4.0/getting-started/browsers-devices/) [BS5](https://getbootstrap.com/docs/5.0/getting-started/browsers-devices/)

> But I would keep it in the footer once logged in

I fully agree.

> Final nitpick: regarding the git log, once done maybe we can try to summarize the changes don here in the commit message ?

I think it might be best for me to figure some stuff out and after discussion archive this PR. Then I'll split up and redo the PRs with good commit messages. I think there are two things we can do; maybe you can help me setup a better and easier develop environment or point me in the right direction. This has my preference, because it'll help me develop future updates more comfortably and learn a lot in the process. The other option is for you to prep the needed files to add Bootswatch 5 in addition to the Bootswatch 3 themes.

Created a little test kit for the login view so you can test it working live.
[loginview-testkit.zip](https://github.com/opensourcepos/opensourcepos/files/6547614/loginview-testkit.zip)

Thanks for taking the time with all the responses.

jekkos commented:
Had a quick look at the testkit, this looks really great, nicely responsive and all. 
I guess if BS5 phases out IE then we just deal with that. I don't think it's worth to spend the time to keep those versions supported if someone is willing to do the effort to upgrade to a new BS. So totally fine by me.

I think the PR is already quite focused in terms of functionality. I just meant that maybe the history log could be cleaned up a little bit before merge.

jekkos commented on bower.json:
```diff
@@ -30,7 +30,8 @@
     "jasny-bootstrap": "~3.1.3",
-    "bootswatch": "3.4.1+1",
+    "bootswatch-3": "bootswatch#3.4.1+1",
+    "bootswatch-5": "bootswatch#5.0.1",
```
Ok I checked the bootswatch repo and they don't have a bower.json for the bootswatch-5, so downloading this version using bower will be a nogo. So what we can do

* Clone the bootswatch repo and add a bower.json ourselves, not sure how much work this would be. We might be able to start from the v3 file and then adapt what we think is needed.
* Migrating to npm for frontend dependencies is a bit out of scope for this PR. But it's something that might need to be prioritized anytime soon
So currently I think we should first try the first option as a quick solution. If this works then we can make a PR to the original repo so we can use the upstream again after.

BudsieBuds commented:
> I guess if BS5 phases out IE then we just deal with that. I don't think it's worth to spend the time to keep those versions supported if someone is willing to do the effort to upgrade to a new BS. So totally fine by me.

I agree, requiring modern browsers is good for security anyways and modern features can be implemented without question.

> I think the PR is already quite focused in terms of functionality. I just meant that maybe the history log could be cleaned up a little bit before merge.

Will do that once it's ready for a merge.

> Ok I checked the bootswatch repo and they don't have a bower.json for the bootswatch-5, so downloading this version using bower will be a nogo. So what we can do
> 
> - Clone the bootswatch repo and add a bower.json ourselves, not sure how much work this would be. We might be able to start from the v3 file and then adapt what we think is needed.
> 
> - Migrating to npm for frontend dependencies is a bit out of scope for this PR. But it's something that might need to be prioritized anytime soon
> So currently I think we should first try the first option as a quick solution. If this works then we can make a PR to the original repo so we can use the upstream again after.

I've been diging into tools like Composer, Bower, NPM and Grunt and I think I've found a good solution. I'll post the final code once I'm done making the SVG and think the code is ready for merging.

BudsieBuds commented:
Finally think it's ready for review. Everything I tested worked perfectly in my environment. Some final changes are downloading of two versions of Bootstrap and Bootswatch through NPM, auto add licenses to the Configuration screen via Grunt and minify login.css, also via Grunt.

I've converted the logo to SVG and made some changes that I hope you like. The mark logo is slightly different from the original; the line width is made the same for the whole logo, as well as the rounding of the mark. Went with the 'flatly'-blue as the primary color, because this is the default theme. Also toyed around with some ideas for a full redesign instead of this refresh, but ended on this. Please check it out and let me know what you think.

jekkos commented:
> Would like to get a return message after the `A database migration to %1 will start after login` message. Something like `Database migration to %1 was succesfull. Please login again`. Is this possible?

I guess this can be done, maybe we should add a ticket for it. If the login page reappears without this message then the migration is done. 

Nice to see you solved the issue by using npm instead. I think this is the way to go and it's nice to see that it can be done for just a couple of dependencies already. This would make the migration path a bit easier as we can then just migrate dependencies as we go (#1891)


To make the review a bit easier and history clean, could you squash all commits into one (maybe using interactive rebase or by reapplying all changes on top of latest master)?

jekkos commented:
@BudsieBuds I have sent you an invite to join  @opensourcepos/org 

If  you join and you push your branch directly on within the project then we can have it deployed automatically on our dev server.

jekkos commented:
> * The other language files have to be updated. Can you do it automatically through Weblate or another way?
> 
> * There seems to be a bug/feature when the reCAPTCHA is set, but not the keys. If I set it with wrong keys and try to login once, it gives an error. The second time it just logs in. This is great if it's designed like this, but does this also happen when the reCAPTCHA is set in the correct way?

The language files can be updated directly by editing the files in the repo or through weblate. 
Did you add new strings in this case? We might need to do a resync to make sure all languages then show them for translation. Once weblate detects these it will notify the contributors for their language.

I haven't myself used the reCaptcha yet, maybe @daN4cat can elaborate on this?

jekkos commented:
> I've converted the logo to SVG and made some changes that I hope you like. The mark logo is slightly different from the original; the line width is made the same for the whole logo, as well as the rounding of the mark. Went with the 'flatly'-blue as the primary color, because this is the default theme. Also toyed around with some ideas for a full redesign instead of this refresh, but ended on this. Please check it out and let me know what you think.

I woulkd say let's try to deploy it on dev and then I'll check it there

daN4cat commented:
reCaptcha works fine with my setup. Now and then the picture selection challenge pops up and clearing the process confirms you are a human. I haven't tested fir a while the negative sequence making mistakes and see what happens. I'll do later.
Also, never tested inputting wrong keys, that error scenario was not covered.

BudsieBuds commented:
> To make the review a bit easier and history clean, could you squash all commits into one (maybe using interactive rebase or by reapplying all changes on top of latest master)?

@jekkos After some trial and error, I think I've done it correctly.

> @BudsieBuds I have sent you an invite to join @opensourcepos/org
> 
> If you join and you push your branch directly on within the project then we can have it deployed automatically on our dev server.

I've joined, but I think you have to merge it, right?

> The language files can be updated directly by editing the files in the repo or through weblate.
> Did you add new strings in this case? We might need to do a resync to make sure all languages then show them for translation. > Once weblate detects these it will notify the contributors for their language.

Yeah, I've added strings and adjusted some. I'm not sure how to best set this up, because I can't see what an admin can see over at Weblate.

When this is deployed on the dev server, I'll go through everything and create issues for things that weren't addressed yet.

jekkos commented:
The issue in the build is that it runs npm 5.6.0 (which is probably pretty old) and your changeset requires 6.14. We might need to upgrade this in the CI configuration then as well.

jekkos commented:
@BudsieBuds What exactly would you want to tweak?
I have given you elevated rights now. 

I checked briefly yesterday for the npm upgrade. It seems that the container we currently use does not have a newer version that supports node 6.x, so not sure how we should proceed. Either we build our own buildbox and add grunt, bower on top of a base nodejs. Just a matter of playing a bit with the Dockerfile.

BudsieBuds commented:
> @BudsieBuds What exactly would you want to tweak?
> I have given you elevated rights now.

I've added some context and images as an example here:
https://translate.opensourcepos.org/translate/opensourcepos/login/en/

At a later time it's probably preferable to switch from PHP strings to GNU gettext for more options, control and fallback to English when a string is not translated. I'll add an issue for this later.

> I checked briefly yesterday for the npm upgrade. It seems that the container we currently use does not have a newer version that supports node 6.x, so not sure how we should proceed. Either we build our own buildbox and add grunt, bower on top of a base nodejs. Just a matter of playing a bit with the Dockerfile.
> 
> We won't be able to deploy your branch until this is resolved though (as nothing is pushed to docker hub yet due to this failure)

Hmm, that's a bummer. Is there anything I can do to help?

jekkos commented:
> > @BudsieBuds What exactly would you want to tweak?
> > I have given you elevated rights now.
> 
> I've added some context and images as an example here:
> https://translate.opensourcepos.org/translate/opensourcepos/login/en/
> 
> At a later time it's probably preferable to switch from PHP strings to GNU gettext for more options, control and fallback to English when a string is not translated. I'll add an issue for this later.
> 
> > I checked briefly yesterday for the npm upgrade. It seems that the container we currently use does not have a newer version that supports node 6.x, so not sure how we should proceed. Either we build our own buildbox and add grunt, bower on top of a base nodejs. Just a matter of playing a bit with the Dockerfile.
> > We won't be able to deploy your branch until this is resolved though (as nothing is pushed to docker hub yet due to this failure)
> 
> Hmm, that's a bummer. Is there anything I can do to help?

Not sure if you're familiar with dockers and stuff? I think we should add a simple repo to the organization's account with one Dockerfile. This file can have a nodejs base image (with newer npm version) and then bower and  grunt can be added on top of hat. Then we can enable the repo in docker hub so it will build a container based on that, which we can use in the travis-ci.

jekkos commented:
see #3208

BudsieBuds commented:
> I've added some context and images as an example here:
> https://translate.opensourcepos.org/translate/opensourcepos/login/en/

Would you be so kind to let me know what you think of this?

> Not sure if you're familiar with dockers and stuff? I think we should add a simple repo to the organization's account with one Dockerfile. This file can have a nodejs base image (with newer npm version) and then bower and grunt can be added on top of that. Then we can enable the repo in docker hub so it will build a container based on that, which we can use in the travis-ci.

Not really, but I tend to learn kind of semi fast-ish. Can I replicate this issue locally and see if I can fix it from there? Should I be able to put my local fork into Docker and see what happens?

jekkos commented:
Once that build finishes, you should try to run this 

```
sed -i 's/digitallyseamless\/nodejs-bower-grunt/opensourcepos\/node-grunt-bower/g' .travis.yml
```
commit the changes and push.. This should trigger the build for your branch and then hopefully the issue will be solved

BudsieBuds commented:
@jekkos Thanks! Unfortunately the build seems to have failed. Don't know much about how this works yet, but this is the error found in the logs:
`The command "echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin" exited with 1.`

jekkos commented:
Yes I spotted that one as well. This might be because the branch is not in the organization's repo, travis might not have these variables configured. Can you push to a new remote branch in this repository ?

```
git remote add github https://github.com/opensourcepos/opensourcepos.git
git push github login-improvements
```

BudsieBuds commented:
@jekkos I don't have permission to do that unfortunately.
```
remote: Permission to opensourcepos/opensourcepos.git denied to BudsieBuds.
fatal: unable to access 'https://github.com/opensourcepos/opensourcepos.git/': The requested URL returned error: 403
```

jekkos commented:
@BudsieBuds Can you try again?

jekkos commented:
you also might need to rebase with master branch

```
git pull --rebase github master
```

BudsieBuds commented:
@jekkos It worked this time, lets see what happens here #3213.

BudsieBuds commented:
It looks so... green 👀.

jekkos commented:
Great! We have it deployed now at https://dev.opensourcepos.org/

@opensourcepos/org check this out!

objecttothis commented:
> 
> 
> Great! We have it deployed now at https://dev.opensourcepos.org/
> 
> @opensourcepos/org check this out!

Very sexy.  Good work @BudsieBuds

jekkos commented:
In master now

izeblood commented:
> > Great! We have it deployed now at https://dev.opensourcepos.org/
> > @opensourcepos/org check this out!
> 
> Very sexy. Good work @BudsieBuds

![loginospos](https://user-images.githubusercontent.com/32444665/134877710-5d226c47-e28d-4fe9-af7d-57488e8f8142.jpg)

objecttothis commented:
Looks good!
